zerver/webhooks/semaphore2/tests.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+from zerver.lib.test_classes import WebhookTestCase
+
+
+class Semaphore2HookTests(WebhookTestCase):
+    STREAM_NAME = 'semaphore2'
+    URL_TEMPLATE = "/api/v1/external/semaphore2?stream={stream}&api_key={api_key}"
+
+    # Messages are generated by Semaphore on git push or a pull request. The subject lines below
+    # contain information on the repo and branch, and the message has details
+    # about the pipeline it triggers.
+
+    def test_semaphore2_push_known_host(self) -> None:
+        expected_topic = u"notifications/rw/webhook_impl"  # repo/branch
+        expected_message = """
+**184065+radwo@users.noreply.github.com** pushed to branch **rw/webhook_impl** of [semaphore/notifications](https://github.com/renderedtext/notifications):
+* **Commit**: [(#2d9f5fc) empty](https://github.com/renderedtext/notifications/commit/2d9f5fcec1ca7c68fa7bd44dd58ec4ff65814563)
+* **Pipeline**:
+  - Name: Notifications
+  - Result: stopped
+* **Blocks**:
+**1. List & Test & Build**:
+  - Result: stopped
+""".strip()
+        self.send_and_test_stream_message('push', expected_topic, expected_message,
+                                          content_type="application/x-www-form-urlencoded")
+
+    def test_semaphore2_push_unknown_host(self) -> None:
+        expected_topic = u"notifications/rw/webhook_impl"  # repo/branch
+        expected_message = """
+**184065+radwo@users.noreply.enterprise.github.com** pushed to branch **rw/webhook_impl** of [semaphore/notifications](https://enterprise.github.com/renderedtext/notifications):
+* **Commit**: [(#2d9f5fc) empty](https://enterprise.github.com/renderedtext/notifications)
+* **Pipeline**:
+  - Name: Notifications
+  - Result: stopped
+* **Blocks**:
+**1. List & Test & Build**:
+  - Result: stopped
+""".strip()
+        self.send_and_test_stream_message('push_unknown_host', expected_topic, expected_message,
+                                          content_type="application/x-www-form-urlencoded")
```

Comment:
Is the `content_type` really `application/x-www-form-urlencoded`? I would expect it to be `application/json`.

Reply:
I also think it should be `application/json`. I just used the existing semaphore integration for adding this integration. I'll change it to `application/json` for semaphore2.